Autopilot/Inc/AttitudePathInterface.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Provides declarations common to the attitude and path manager interfaces.
+ * Author: Anthony Bertnyk
+ */
+
+#ifndef ATTITUDE_PATH_INTERFACE_HPP
+#define ATTITUDE_PATH_INTERFACE_HPP
+
+extern "C"
+{
+#include "cmsis_os.h"
+}
+
+//Commands for path manager to send to attitude manager.
+typedef struct CommandsForAM{
+	float roll,pitch,yaw;	// commanded orientation (radians)
+	float airspeed;			// commanded airspeed (m/s)
+} CommandsForAM;
+
+//Data for attitude manager to send to path manager.
+typedef struct AttitudeData{
+	float roll,pitch,yaw;	// current orientation (radians)
+	float airspeed;			// current airspeed (m/s)
+	float accX, accY, accZ; // accelerometer data
+	float gyrX, gyrY, gyrZ; // gyroscope data
+} AttitudeData;
+
+const char MAIL_Q_SIZE = 1;
+
+//Set up a mail queue for sending commands to the attitude manager
+extern osMailQDef(commandsMailQ, MAIL_Q_SIZE, CommandsForAM);
+extern osMailQId commandsMailQ;
+
+
+//Set up a mail queue for sending data to the path manager
+extern osMailQDef(attitudeDataMailQ, MAIL_Q_SIZE, AttitudeData);
+extern osMailQId attitudeDataMailQ;
```

Comment:
Both the path manager and attitude manager need to use both these mail queues to communicate. I'm using the CMSIS library to declare them here but the unit testing system is unaware of any RTOS stuff. Is there a better way to let both sides of the interface access the mail queues?

Reply:
maybe, but this is the most intuitive way. What I suggest you do is add a define in CmakeLists.txt in the targetBuild section. Then what you can do is wrap up all the Freertos specific stuff in this file inside a #ifdef TARGET_BUILD. Something like this already exists for unit test builds.